File: 백준/Silver/2615. 오목/오목.py
import sys
input = sys.stdin.readline

board = [list(map(int, input().split())) for _ in range(19)]

# → ↓ ↗ ↘
dx = [1, 0, 1, 1]  
dy = [0, 1, -1, 1]  

for y in range(19):
    for x in range(19):
        if board[y][x] != 0:
            color = board[y][x]
            for i in range(4):
                nx, ny, cnt = x + dx[i], y + dy[i], 1
                while cnt < 5 and 0 <= nx < 19 and 0 <= ny < 19 and board[ny][nx] == color:
                    cnt += 1
                    nx, ny = nx + dx[i], ny + dy[i]

                if cnt == 5: # 육목
                    prev_x, prev_y = x - dx[i], y - dy[i]
                    next_x, next_y = nx, ny

                    if (0 <= prev_x < 19 and 0 <= prev_y < 19 and board[prev_y][prev_x] == color):
                        continue
                    if (0 <= next_x < 19 and 0 <= next_y < 19 and board[next_y][next_x] == color):
                        continue

                    print(color)
                    print(y + 1, x + 1)
                    exit(0)

print(0)

File: 백준/Silver/2615. 오목/오목.cc
#include <iostream>
#include <vector>

using namespace std;

int main()
{
	vector<vector<int>> board(19, vector<int>(19, 0));
	for (int i = 0; i < 19; i++)
		for (int j = 0; j < 19; j++)
			cin >> board[i][j];

	// → ↓ ↘ ↗
	int dx[4] = { 1, 0, 1, 1 };
	int dy[4] = { 0, 1, 1, -1 };

	for (int y = 0; y < 19; y++)
	{
		for (int x = 0; x < 19; x++)
		{
			if (board[y][x] == 0)
				continue;
			int color = board[y][x];
			for (int i = 0; i < 4; i++) // → ↓ ↘ ↗
			{
				int cnt = 1;
				int nx = x + dx[i], ny = y + dy[i];
				while (cnt < 5 
					&& 0 <= nx && nx < 19
					&& 0 <= ny && ny < 19)
				{
					if (board[ny][nx] != color)
						break;
					cnt++;
					nx = x + dx[i] * cnt;
					ny = y + dy[i] * cnt;
				}

				if (cnt < 5)
					continue;

				// 육목 체크
				if (0 <= nx && nx < 19
					&& 0 <= ny && ny < 19)
				{
					if (board[ny][nx] == color)
						continue;
				}

				int prev_x = x - dx[i], prev_y = y - dy[i];
				if (0 <= prev_x && prev_x < 19
					&& 0 <= prev_y && prev_y < 19)
				{
					if (board[prev_y][prev_x] == color)
						continue;
				}

				// 오목
				cout << color << "\n";
				cout << y + 1 << " " << x + 1 << "\n";
				return 0;
			}
		}
	}

	cout << 0 << "\n";
	return 0;
}